test/distributed/org/apache/cassandra/distributed/test/tracking/MutationTrackingWriteForwardingTest.java

```diff
@@ -82,56 +77,44 @@ public void testBasicWriteForwarding() throws Throwable
-                // replicas (until background reconciliation broadcast is implemented), and ignored on others.
-                Set<IInvokableInstance> leaderOrNonReplica = new HashSet<>(2);
-                Set<IInvokableInstance> replicas = new HashSet<>();
-                for (IInvokableInstance instance : cluster)
-                {
-                    int unreconciled = instance.callOnInstance(() -> {
-                        Token token = DatabaseDescriptor.getPartitioner().getMinimumToken();
-                        Range<Token> fullRange = new Range<>(token, token);
-                        TableId tableId = Schema.instance.getTableMetadata(keyspaceName, tableName).id;
-                        MutationSummary summary = MutationTrackingService.instance.createSummaryForRange(fullRange, tableId, true);
-                        return summary.unreconciledIds();
-                    });
-                    int lastUnreconciled = instanceUnreconciled.getOrDefault(instance, 0);
-                    int newUnreconciled = unreconciled - lastUnreconciled;
+            Thread.sleep(1000); // allow time for all offsets to be broadcasted
 
-                    if (newUnreconciled == 0)
-                    {
-                        // instance already reconciled (as leader) or did not receive new mutation ID (non-replica)
-                        leaderOrNonReplica.add(instance);
-                    }
-                    else if (newUnreconciled == 1)
-                    {
-                        // instance has not reconciled, so it's a replica (until reconciliation broadcast is implemented)
-                        replicas.add(instance);
-                    }
-                    else
-                    {
-                        Assertions.fail("Should not have more than one new unreconciled mutation");
-                    }
-                    instanceUnreconciled.put(instance, unreconciled);
-                }
-                Assertions.assertThat(leaderOrNonReplica).hasSize(2);
-                Assertions.assertThat(replicas).hasSize(TOTAL_RF - 1);
+            int allReconciled = 0;
+            int allUnreconciled = 0;
+
+            // Writes should be ack'd in the journal too, but these could lag behind client acks, so could be
+            // permissive here. Each write should be reconciled everywhere.
+            for (IInvokableInstance instance : cluster)
+            {
+                int reconciled = instance.callOnInstance(() -> {
+                    Token token = DatabaseDescriptor.getPartitioner().getMinimumToken();
+                    Range<Token> fullRange = new Range<>(token, token);
+                    TableId tableId = Schema.instance.getTableMetadata(keyspaceName, tableName).id;
+                    MutationSummary summary = MutationTrackingService.instance.createSummaryForRange(fullRange, tableId, true);
+                    return summary.reconciledIds();
```

Comment:
Note for later - would be nice to have metrics to differentiate writes that are reconciled locally on the leader, from in-band write acknowledgements on replicas, or from broadcast.